hawk/app/lib/util.rb

```diff
@@ -363,15 +363,43 @@ def acl_version
   end
   module_function :acl_version
 
-  def get_meta_data(agent)
-    res = safe_x("/usr/sbin/crm_resource", "--show-metadata", agent)
+  # Hash#dig from Ruby 2.3
+  # https://apidock.com/ruby/Hash/dig
+  def dig_hash(hash, *keys)
+    res = hash
+    keys.each do |key|
+      res = res[key]
+      return nil if res.nil?
+    end
+    res
+  end
+  module_function :dig_hash
+
+  def get_metadata_hash(agent)
     sub_map = { 'type="string"' => 'type="text"',
                 '(type="boolean".*)default="(yes|1)"' => '\1default="true"',
                 '(type="boolean".*)default="(no|0)"' => '\1default="false"' }
-    sub_map.each { |k,v| res.gsub!(/#{k}/i, v) }
+
+    xml = safe_x("/usr/sbin/crm_resource", "--show-metadata", agent)
+    sub_map.each { |k,v| xml.gsub!(/#{k}/i, v) }
+    res = Hash.from_xml(xml)
+
+    # For stonith agents, complete the instance attributes with the ones that
```

Comment:
@petrpavlu  I would personaly segregate responsability of the function. LIke  for the stonith_agents I would create an separate function wich is called by the main get_metadata function